tests/func/test_remote.py

```diff
@@ -47,7 +48,7 @@ def test_relative_path(self):
         # dir path written to config should be just one level above.
         rel = os.path.join("..", dname)
         config = configobj.ConfigObj(self.dvc.config.files["repo"])
-        self.assertEqual(config['remote "mylocal"']["url"], rel)
+        self.assertEqual(Path(config['remote "mylocal"']["url"]), Path(rel))
```

Comment:
Let's just
```
import posixpath
...
rel = posixpath.join("..", dname)
...
self.assertEqual(config['remote "mylocal"']["url"], rel)
```
just to make it more explicit :slightly_smiling_face:

Reply:
Same in the tests above

Reply:
How about 

```
def test_relative_path(self):
        dname_parts = ["..", "path", "to", "dir"]
        dname = posixpath.join(*dname_parts)
        ret = main(["remote", "add", "mylocal", dname])
        self.assertEqual(ret, 0)

        # NOTE: we are in the repo's root and config is in .dvc/, so
        # dir path written to config should be just one level above.
        rel = poxispath.join("..", *dname_parts)
        config = configobj.ConfigObj(self.dvc.config.files["repo"])
        self.assertEqual(Path(config['remote "mylocal"']["url"]), Path(rel))
```

instead? That way we still test passing in a Windows path and getting out a posix one.

Reply:
Oops, sorry, didn't notice that `dname` is joined above. So I suppose you really meant:
```
        dname_parts = ["..", "path", "to", "dir"]
        dname = os.path.join(*dname_parts)
        ...
        rel = poxispath.join("..", *dname_parts) 
```
right? This actually qualifies as a func test that makes the unit tests we have below redundant :slightly_smiling_face: Maybe consider just keeping these instead.

Reply:
Oops, you're right of course. Re the other tests, do you mean the others in this PR? If so, I think it's worth keeping them since they test relative, absolute and remote paths on both windows and linux, whereas this test is only for a relative path. Or did you mean pre-existing ones in this file?